pkg/controllers/workapplier/status.go

```diff
@@ -151,24 +148,27 @@ func (r *Reconciler) refreshWorkStatus(
 		if bundle.availabilityResTyp == ManifestProcessingAvailabilityResultTypeNotTrackable {
 			untrackableAppliedObjectsCount++
 		}
+		if isManifestObjectDiffReported(bundle.reportDiffResTyp) {
+			diffReportedObjectsCount++
+		}
 	}
 
 	// Refresh the Work object status conditions.
 
 	// Do a sanity check.
-	if appliedManifestsCount > manifestCount || availableAppliedObjectsCount > manifestCount {
+	if appliedManifestsCount > manifestCount || availableAppliedObjectsCount > manifestCount || untrackableAppliedObjectsCount > manifestCount || diffReportedObjectsCount > manifestCount {
 		// Normally this should never happen.
 		return controller.NewUnexpectedBehaviorError(
-			fmt.Errorf("the number of applied manifests (%d) or available applied objects (%d) exceeds the total number of manifests (%d)",
-				appliedManifestsCount, availableAppliedObjectsCount, manifestCount))
+			fmt.Errorf("the number of applied manifests (%d), available applied objects (%d), untrackable applied objects (%d), or diff reported objects (%d) exceeds the total number of manifests (%d)",
+				appliedManifestsCount, availableAppliedObjectsCount, untrackableAppliedObjectsCount, diffReportedObjectsCount, manifestCount))
 	}
 
 	if work.Status.Conditions == nil {
 		work.Status.Conditions = []metav1.Condition{}
 	}
 	setWorkAppliedCondition(work, manifestCount, appliedManifestsCount)
-	setWorkAvailableCondition(work, manifestCount, availableAppliedObjectsCount, untrackableAppliedObjectsCount)
-	setWorkDiffReportedCondition(work, manifestCount, diffedObjectsCount)
+	setWorkAvailableCondition(work, manifestCount, appliedManifestsCount, availableAppliedObjectsCount, untrackableAppliedObjectsCount)
```

Comment:
Nit: what if the appliedCondition (say false) does not agree with appliedManifestsCount (say == manifestCount)? I think it's slightly better to use the output of the setWorkAppliedCondition to determine the appliedCondition than using "appliedManifestsCount" as they are two source of truth